From 70cc2a7cef05b98e371f414b8f6452a9312ca4f7 Mon Sep 17 00:00:00 2001
From: "Chen, Gang G" <gang.g.chen@intel.com>
Date: Wed, 6 Dec 2023 03:32:00 +0800
Subject: [PATCH 48/54] Set device as unlocked state by default on userdebug

This is only WA patch to set device as unlocked.
The patch should be removed if fastboot is supported
for BM.

With this patch, device will be unlocked on userdebug build;
For user build, it still locked.

Tracked-On: OAM-108853
Signed-off-by: Chen, Gang G <gang.g.chen@intel.com>
Author: Chen, Gang G <gang.g.chen@intel.com>
---
 kernelflinger.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/kernelflinger.c b/kernelflinger.c
index ec08952..c411798 100644
--- a/kernelflinger.c
+++ b/kernelflinger.c
@@ -1523,11 +1523,17 @@ EFI_STATUS efi_main(EFI_HANDLE image, EFI_SYSTEM_TABLE *sys_table)
 
 	need_lock = device_need_locked();
 
+#ifndef USER
+	/* WA patch to set device as unlocked by default for userdebug build
+	 */
+	set_current_state(UNLOCKED);
+#else
 	/* For civ, flash images to disk is not MUST. So set device to LOCKED
 	 * state by default on the first boot.
 	*/
 	if (need_lock)
 		set_current_state(LOCKED);
+#endif
 
 	ret = set_device_security_info(NULL);
 	if (EFI_ERROR(ret)) {
-- 
2.25.1

